Update kubejs examples for recipes

Titop54 opened the pull request:
Added more info and changes according to https://discord.com/channels/1071821610836828200/1305926116929699900/1416688948985593936

Titop54 committed:
Update kubejs examples for recipes

alec016 commented on kubejsExample/server_scripts/Recipes/basic_recipes.js:
```diff
@@ -43,15 +60,16 @@ If 2 arguments are passed -> Item you specified with Z% and position 0,0
     event.recipes.modular_machinery_reborn.machine_recipe(machine_id, time)
     .requireItem("minecraft:spruce_boat", 10, 10)
-    .produceItem("minecraft:oak_log", 0.1, 40, 10)
+    .produceItem("minecraft:oak_log", 0.1, 40, 10) //has a 1% to be produce
```
actually, this is a 10% chance, not 1%

alec016 commented on kubejsExample/server_scripts/Recipes/functions.js:
```diff
@@ -5,14 +5,37 @@
-.requireFunctionOnStart("id") -> Useful to check items to give a boost
-.requireFunctionEachTick("id") -> When you want to check item to modify recipe time or stop it
-.requireFunctionOnEnd("id") -> Decides if you get a result or not
+.requireFunctionToStart("id", []) -> Decides if you can start a recipe
```
args are not required, also args can be passed directly as arguments, it will get it as an array internally

alec016 commented on kubejsExample/server_scripts/Recipes/functions.js:
```diff
@@ -5,14 +5,37 @@
+
+- Or passing data to the function itself, as an array of text
+
+.requireFunctionOnEachTick("boat_chooser", ["hello", "im", "inside", "the boat"])
```
`.requireFunctionOnEachTick("boat_chooser", "hello", "im", "inside", "the boat")` this should be enough

alec016 commented on kubejsExample/server_scripts/Recipes/functions.js:
```diff
@@ -23,19 +46,50 @@ ServerEvents.recipes(event => {
     .requireItem("minecraft:birch_boat")
     .produceItem("minecraft:oak_log")
-    .requireFunctionOnEachTick("boat_chooser")
+    .requireFunctionOnEachTick("boat_chooser", [])
```
`.requireFunctionOnEachTick("boat_chooser")` to dont pass any args

Titop54 committed:
Forgot a 0

Titop54 committed:
Extra argument

Titop54 committed:
change the small example

alec016 commented on kubejsExample/server_scripts/Recipes/functions.js:
```diff
@@ -23,19 +61,51 @@ ServerEvents.recipes(event => {
     event.setBaseSpeed(2) //min is 0.1
 
+    let data = event.getContext().args //to get the data passed
+    let size_data = event.getContext().args.size()
```
`.getContext()` returns itself, so its redundant

Titop54 committed:
Update functions.js

alec016 commented:
i drop the changelog here, so you can see all the changes, those changes includes progress positioning, sound customizability, new requirements(added all the available methods for each requirement in the changelog)
[CHANGELOG.md](https://github.com/user-attachments/files/24100511/CHANGELOG.md)

Titop54 committed:
Added more information on some parts lacking them (like URL to useful information like Javadoc 25 or more valid options)

New content from 3.0.5 and lower, like the new requirements or methods

Added new tips on doing recipes

Changed some old progress bar (still mantained the .progressX() and .progressY() on the comments

Changed some comments on position (from the same like to above or using multiline instead of multiple of 1 line)

alec016 commented on kubejsExample/server_scripts/Structures/modifiers.js:
maybe adding the new controller model by status will be a good idea

alec016 commented on kubejsExample/server_scripts/Structures/more.js:
all the hatches can change textures, including the new ones.

Including ars and mekanism addons

alec016 commented on kubejsExample/server_scripts/Structures/more.js:
```diff
@@ -0,0 +1,107 @@
+.placeStructure(MMRStructureBuilder.createRequirement()) // Same as above with true
+
+It's similar to the structure on the .structure(), but needs the special hatches, and m (the controller) is changed to $
+You need to use the Structure Creator
```
the not structure/not controller is the same chartacter `_`

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
 For the other blocks, you have the following (needs the block):
-- chunkload -> Allows to chunkload X number of chunks, if another 2 more arguments,
+- .chunkload -> Allows to chunkload X number of chunks, if another 2 more arguments,
                puts X and Y text on recipe viewer
```
no longer needed x, y

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
+- .biomes -> Allows to specify if you can run a recipe in certain biomes or not
             Needs a list (["biome_here", "biome_here"]) of biomes
             If passed after the list a true, then, those biome becomes a blacklist
             If passed 3 arguments or 4, the last 2 are considered the X and Y on JEI
```
no x, y arguments anymore

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
-- weather -> Allows to run a recipe when on a specific weather ("rain", "clear", "snow", "thunder")
-             If passed another 2 arguments, the last 2 are considered the X and Y on Jei
+- .weather -> Allows to run a recipe when on a specific weather ("rain", "clear", "snow", "thunder")
+              If passed another 2 arguments, the last 2 are considered the X and Y on Jei
```
no position arguments anymore

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
-- time -> Allows you to specify a range where recipe can be done (it's relative, from 0 to 24000), more info here https://wikis.degrassi.es/docs/modular-machinery-reborn/section/misc/article/range
-          If passed another 2 arguments, the last 2 are considered the X and Y on Jei
+- .time -> Allows you to specify a range where recipe can be done (it's relative, from 0 to 24000), more info here: https://wikis.degrassi.es/docs/modular-machinery-reborn/section/misc/article/range
+           If passed another 2 arguments, the last 2 are considered the X and Y on Jei
```
no position arguments anymore

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
+
+Similar to a Furnace burning fuel:
+.requireFuel(amount) -> Using the fuel hatch to do a recipe
+.requireFuel(amount, data) -> data is similar to ProgressData, except using FuelData and direction is inverted (instead of left to right, from right to left for example)
```
has x and y arguments too

alec016 commented on kubejsExample/server_scripts/Recipes/more_complex_recipies.js:
```diff
@@ -5,37 +5,106 @@
+command is a String, it must include the /, similar on how you use it on the chat
+permissionLevel uses from 0 to 4 (both inclusive, default is 2) (More info here: https://minecraft.wiki/w/Permission_level)
+log is a true or false, telling you if it should log the command (false by default)
+
```
commands must start with `/`

alec016 commented on kubejsExample/server_scripts/Structures/more_complex_structure.js:
```diff
@@ -131,3 +131,43 @@ MMREvents.machines(event => {
+Well, you can do that too!
+
+You can use a block or a tag:
+
```
i think should be better to specify that the ranges must be numbers

Titop54 committed:
Changes

Titop54 committed:
Example of requireFuel

alec016 commented on kubejsExample/server_scripts/Recipes/tips.js:
```diff
@@ -0,0 +1,116 @@
+        event.recipes.modular_machinery_reborn.machine_recipe(machine_id, time)
+            .requireItem(copy, 10, 10)
+            .emptyItem(10, 20)
+            .produceItem(`${number}x minecraft:oak_log`, 0.1, 40, 10) //Up to 90 items
```
up to 99 items

alec016 commented on kubejsExample/server_scripts/Structures/more.js:
```diff
@@ -0,0 +1,127 @@
+Mekanism:
+modular_machinery_reborn_mekanism:chemical_input_hatch_<size>
+modular_machinery_reborn_mekanism:chemical_output_hatch_<size>
+modular_machinery_reborn_mekanism:heat_input_vent<_size>
```
heat doesnt have size

Titop54 committed:
Changes

Titop54 committed:
Forgot a progressX and progressY on functions.js

alec016 commented on kubejsExample/server_scripts/Structures/modifiers.js:
```diff
@@ -29,6 +31,7 @@ MMREvents.machines(event => {
     Can be on assets/mod/model (example: minecraft:block/grass) or a blockid (modid:blockid)
+    ControllerModel is accesible without adding a Java.loadClass
     */
     .controllerModel(ControllerModel.of('minecraft:stone'))
```
model by machine status is already published

Titop54 committed:
Added a detail

alec016 commented on kubejsExample/server_scripts/Structures/more_complex_structure.js:
```diff
@@ -131,3 +131,72 @@ MMREvents.machines(event => {
+            "a":["modular_machinery_reborn:casing_plain"]
+        })
+    )
+    .controllerModel(ControllerModel.of("immersiveengineering:refinery"))
```
Could you add model by states

Titop54 committed:
Merge branch 'alec016:1.21-NeoForge' into 1.21-NeoForge

Titop54 committed:
Merge branch 'alec016:1.21-NeoForge' into 1.21-NeoForge

Titop54 committed:
Tooltips
